pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -97,5 +117,174 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
+	completed := condition.IsTrue(instance.Status.Conditions, toolchainv1alpha1.UserSignupComplete)
+
+	// If the status is not yet provisioning or completed, check for automatic or manual approval
+	if !provisioning && !completed {
+		userApprovalPolicy, err := r.ReadUserApprovalPolicyConfig()
+		if err != nil {
+			err := r.setStatusFailedToReadUserApprovalPolicy(instance, "Failed to read ConfigMap value " + config.ToolchainConfigMapUserApprovalPolicy)
+			if err != nil {
+				reqLogger.Error(err, "Error updating UserSignup Status", "UserID", instance.Spec.UserID)
+				return reconcile.Result{}, err
+			}
+
+			reqLogger.Error(err, "Error reading user approval policy")
+			return reconcile.Result{}, err
+		}
+
+		// If the signup has been explicitly approved (by an admin), or the user approval policy is set to automatic,
+		// then proceed with the signup
+		if instance.Spec.Approved || userApprovalPolicy == config.UserApprovalPolicyAutomatic {
+			var targetCluster string
+
+			// If a target cluster hasn't been selected, select one from the members
+			if instance.Spec.TargetCluster != "" {
+				targetCluster = instance.Spec.TargetCluster
+			} else {
+				// Automatic cluster selection
+				members := cluster.GetMemberClusters()
+				if len(members) > 0 {
+					targetCluster = members[0].Name
+				} else {
+					err := r.setStatusNoClustersAvailable(instance, "No member clusters found")
+					if err != nil {
+						reqLogger.Error(err, "Error updating UserSignup Status", "UserID", instance.Spec.UserID)
+						return reconcile.Result{}, err
+					}
+
+					err = NewSignupError("No target clusters available")
+					reqLogger.Error(err, "No member clusters found")
+					return reconcile.Result{}, err
+				}
+			}
+
+			// Set the "Provisioning" status condition
+			err := r.setStatusProvisioning(instance, "Provisioning User")
+			if err != nil {
+				reqLogger.Error(err, "Error updating UserSignup Status", "UserID", instance.Spec.UserID)
+				return reconcile.Result{}, err
+			}
+
+			// Provision the MasterUserRecord
+			err = r.provisionMasterUserRecord(instance, targetCluster, reqLogger)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+	}
+
 	return reconcile.Result{}, nil
 }
+
+// provisionMasterUserRecord does the work of provisioning the MasterUserRecord
+func (r *ReconcileUserSignup) provisionMasterUserRecord(userSignup *toolchainv1alpha1.UserSignup, targetCluster string, logger logr.Logger) error {
+	userAccounts := []toolchainv1alpha1.UserAccountEmbedded{
+		{
+			TargetCluster: targetCluster,
+			Spec: toolchainv1alpha1.UserAccountSpec{
+				NSTemplateSet: toolchainv1alpha1.NSTemplateSetSpec{
+					Namespaces: []toolchainv1alpha1.Namespace{},
+				},
+			},
+		},
+	}
+
+	mur := &toolchainv1alpha1.MasterUserRecord{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: userSignup.Spec.UserID,
+			Namespace: config.GetOperatorNamespace(),
```

Comment:
You can use the namespace name from the request - both resources should be in the same namespace

Reply:
Fixed to use the same namespace as the `UserSignup` instance.  I've also made the changes to update the status in case of a MUR that already exists, I'll just give Alexey a little more time to respond to my previous comment, if I don't hear from him tomorrow morning my time I'll just go ahead with the change.